itest/litd_custom_channels_test.go

```diff
@@ -2162,16 +2268,41 @@ func testCustomChannelsLiquidityEdgeCases(ctx context.Context,
 	})
 	require.NoError(t.t, err)
 
+	opts = []payOpt{
+		withPayErrSubStr("context deadline exceeded"),
+		withFailure(lnrpc.Payment_FAILED, failureNone),
+	}
+	addGroupModeOpt(&opts, groupMode, groupID)
+
 	// Now Erin tries to pay the invoice. Since rfq quote cannot satisfy the
 	// total amount of the invoice this payment will fail.
 	payInvoiceWithSatoshi(
-		t.t, erin, iResp, withPayErrSubStr("context deadline exceeded"),
-		withFailure(lnrpc.Payment_FAILED, failureNone),
+		t.t, erin, iResp, opts...,
 	)
 
 	logBalance(t.t, nodes, assetID, "after small manual rfq")
 }
 
+// testCustomChannelsLiquidityEdgeCases is a test that runs through some
+// taproot asset channel liquidity related edge cases.
+func testCustomChannelsLiquidityEdgeCases(ctx context.Context,
+	net *NetworkHarness, t *harnessTest) {
+
+	// Run liquidity edge cases and only use single asset IDs for invoices
+	// and payments.
+	testCustomChannelsLiquidtyEdgeCasesCore(ctx, net, t, false)
+}
+
+// testCustomChannelsLiquidityEdgeCasesGroup is a test that runs through some
+// taproot asset channel liquidity related edge cases using group keys.
+func testCustomChannelsLiquidityEdgeCasesGroup(ctx context.Context,
+	net *NetworkHarness, t *harnessTest) {
+
+	// Run liquidity edge cases and only use group keys for invoices and
+	// payments.
+	testCustomChannelsLiquidtyEdgeCasesCore(ctx, net, t, true)
```

Comment:
Can't we run this directly in `testCustomChannelsLiquidityEdgeCases`? Meaning to call the same sub test function there once with true once with false.
Each new test has quite a bit of overhead, so the more we can combine, the better.

Reply:
Yeah, I've actually come full circle now and started taking apart the HTLC force close test in https://github.com/lightninglabs/lightning-terminal/pull/1017 and making two out of them. That helps with starting with a clean slate.
And the performance optimizations I have in https://github.com/lightninglabs/lightning-terminal/pull/987 should make it so that there is no additional overhead for running things in the same test vs. in separate ones.
So this is okay the way it is now.